src/lib/deploy.js

```diff
@@ -610,6 +691,80 @@ export function chooseSmartContract(config, deploy, deployAliasName) {
   return '';
 }
 
+// Finds the the user defined name argument of the ZkProgram that is verified in a smart contract
+// https://github.com/o1-labs/o1js/blob/7f1745a48567bdd824d4ca08c483b4f91e0e3786/src/examples/zkprogram/program.ts#L16.
+function getZkProgramNameArg(message) {
+  let zkProgramNameArg = null;
+
+  // Regex to parse the ZkProgram name argment that is specified in the given message
+  const regex =
+    /depends on ([\w-]+), but we cannot find compilation output for ([\w-]+)/;
+  const match = message.match(regex);
+  if (match && match[1] === match[2]) {
+    zkProgramNameArg = match[1];
+    return zkProgramNameArg;
+  }
+  return zkProgramNameArg;
+}
+
+/**
+ * Find the file and variable name of the ZkProgram.
+ * @param {string}    buildPath    The glob pattern--e.g. `build/**\/*.js`
+ * @param {string}    zkProgramNameArg The user-specified ZkProgram name argument https://github.com/o1-labs/o1js/blob/7f1745a48567bdd824d4ca08c483b4f91e0e3786/src/examples/zkprogram/program.ts#L16.
+ * @returns {Promise<{zkProgramVarName: string, zkProgramFile: string}>}
+ *      An object containing the variable name (`zkProgramVarName`)
+ *      of the ZkProgram and the file name (`zkProgramFile`) in which the specified ZkProgram is found.
+ *      Returns null if the ZkProgram is not found.
+ */
+async function findZkProgramFile(buildPath, zkProgramNameArg) {
+  if (process.platform === 'win32') {
+    buildPath = buildPath.replaceAll('\\', '/');
+  }
+  const files = await glob(buildPath);
+
+  for (const file of files) {
+    const zkProgram = fs.readFileSync(file, 'utf-8');
+
+    // Regex is used to find and extract the variable name of the ZkProgram
+    // that has a matching name argument that is verified in the smart contract
+    // to be deployed.
+    const regex =
+      /(\w+)\s*=\s*ZkProgram\(\{[\s\S]*?name:\s*'([\w-]+)'[\s\S]*?\}\);/g;
+    let match;
+
+    while ((match = regex.exec(zkProgram)) !== null) {
+      const [_, zkProgramVarName, nameArg] = match;
+
+      if (nameArg === zkProgramNameArg) {
+        return { zkProgramVarName, zkProgramFile: path.basename(file) };
+      }
+    }
+  }
+}
+
+/**
+ * Find and import the ZkProgram.
+ * @param {string}    projectRoot    The root directory path of the smart contract
+ * @param {string}    zkProgramNameArg The user-specified ZkProgram name argument https://github.com/o1-labs/o1js/blob/7f1745a48567bdd824d4ca08c483b4f91e0e3786/src/examples/zkprogram/program.ts#L16.
+ * @returns {Promise<object>}          The ZkProgram.
+ *
+ */
+async function getZkProgram(projectRoot, zkProgramNameArg) {
+  let { zkProgramFile, zkProgramVarName } = await findZkProgramFile(
```

Comment:
I'm getting the following error while trying to deploy https://github.com/Raunaque97/RepeatingLifeZK/tree/main/Contracts/src

```shell
zk deploy
✔ Which deploy alias do you want to deploy to? · lightnet3
✔ Build project
✔ Generate build.json
✔ Choose smart contract
  Only one smart contract exists in the project: GameOfLife
  Your config.json was updated to always use this
  smart contract when deploying to this deploy alias.
✖ Generate verification key (takes 10-30 sec)
  TypeError: Cannot destructure property 'zkProgramFile' of '(intermediate value)' as it is undefined.
TypeError: Cannot destructure property 'zkProgramFile' of '(intermediate value)' as it is undefined.
    at getZkProgram (file:///Users/serhii/projects/o1labs/zkapp-cli/src/lib/deploy.js:753:9)
    at async generateVerificationKey (file:///Users/serhii/projects/o1labs/zkapp-cli/src/lib/deploy.js:548:19)
    at async file:///Users/serhii/projects/o1labs/zkapp-cli/src/lib/deploy.js:271:7
    at async step (file:///Users/serhii/projects/o1labs/zkapp-cli/src/lib/helpers.js:17:20)
    at async deploy (file:///Users/serhii/projects/o1labs/zkapp-cli/src/lib/deploy.js:267:41)
    at async Object.handler (file:///Users/serhii/projects/o1labs/zkapp-cli/src/bin/index.js:127:21)
```

Reply:
The I only used subset of files:

```shell
	src/gameOfLife.ts
	src/gameOfLifeSimulator.ts
	src/gameOfLifeZkProgram.ts
	src/run.ts
```

Reply:
I believe the issue is that there is no `name` property in that ZkProgram.

Reply:
Actually it seems the initial issue was because of outdated experimental namespace in use.

Reply:
And absence of the `name` property of ZkProgram.
Probably API have changed.

Reply:
Successfully deployed!